src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,401 @@
+                break;
+        }
+    }
+    if(hipThreadIdx_x == 0)
+    {
+        if(beginResult == srcLength || endResult == 0)
+        {
+            beginTensor[id_z] = 0;
+            lengthTensor[id_z] = 0;
+        }
+        else
+        {
+            int detectBegin = beginResult;
+            int detectEnd = endResult - beginResult + 1;
+            if(detectBegin != 0 && detectEnd != 0)
+            {
+                int newBegin = max(detectBegin - (windowLength - 1), 0);
+                detectEnd += detectBegin - newBegin;
+                detectBegin = newBegin;
+            }
+            beginTensor[id_z] = detectBegin;
+            lengthTensor[id_z] = detectEnd;
+        }
+    }
+}
+
+// -------------------- Set 4 -  host helpers for kernel executor --------------------
+
+int prev_pow2(int n)
+{
+    int pow2 = 1;
+    while (n - pow2 > pow2)
+        pow2 += pow2;
+
+    return pow2;
+}
+
+int next_pow2(int n)
+{
+    int pow2 = 1;
+    while (n > pow2)
+        pow2 += pow2;
+
+    return pow2;
+}
+
+// -------------------- Set 5 - non silent region kernels executor --------------------
+
+RppStatus hip_exec_non_silent_region_detection_tensor(Rpp32f *srcPtr,
+                                                      RpptDescPtr srcDescPtr,
+                                                      Rpp32s *srcLengthTensor,
+                                                      Rpp32s *detectedIndexTensor,
+                                                      Rpp32s *detectionLengthTensor,
+                                                      Rpp32f cutOffDB,
+                                                      Rpp32s windowLength,
+                                                      Rpp32f referencePower,
+                                                      Rpp32s resetInterval,
+                                                      rpp::Handle& handle)
+{
+    // allocate temporary memory for MMS Array
+    Rpp32f *mmsArr;
+    hipMalloc(&(mmsArr), srcDescPtr->n * srcDescPtr->strides.nStride * sizeof(Rpp32f));
+
+    int maxSharedMemoryInBytes = handle.GetLocalMemorySize();
+    int maxSharedMemoryElements = maxSharedMemoryInBytes / sizeof(Rpp32f);
+    int kSharedMemBanks = 32;
+    int inputTileLength = prev_pow2(maxSharedMemoryElements * kSharedMemBanks / (kSharedMemBanks + 1));
+
+    if (resetInterval > 0 && resetInterval < inputTileLength)
+    {
+        auto p = prev_pow2(resetInterval);
+        auto n = next_pow2(resetInterval);
+        if (p > windowLength)
+            inputTileLength = p;
+        else if (n < inputTileLength)
+            inputTileLength = n;
+    }
+
+    int sharedMemorySizeInBytes = smem_pos(inputTileLength) * sizeof(Rpp32f);
+    int outputTileLength = inputTileLength - windowLength;
+    float windowFactor = 1.0f / windowLength;
+
+    if (outputTileLength <= 0)
+    {
+        std::cout << "Invalid output tile length! " << std::endl;
+        return RPP_ERROR;
```

Comment:
Adding error codes
But instead of moving this code  where we compute output tile length and shared memory size  to api file
I have kept in kernel file itself and stored the return status of hip executor in api file
https://github.com/sampath1117/rpp/blob/sr/nsr_hip/src/modules/rppt_tensor_audio_augmentations.cpp#L299

Please let me know if i need to change this approach here